tempesta_fw/http_parser.c

```diff
@@ -3714,20 +3211,19 @@ tfw_http_parse_req(void *req_data, unsigned char *data, size_t len,
 			__FSM_MOVE_f(Req_UriAbsPath, &req->uri_path);
 		}
 
-		if (parser->_i_st == I_0) {
+		if (!parser->_i_st)
 			TRY_STR_INIT();
-			parser->_i_st = Req_I_UriMarkStart;
-		}
 		__fsm_sz = __data_remain(p);
 		__fsm_n = __parse_uri_mark(req, p, __fsm_sz);
 		if (__fsm_n == CSTR_POSTPONE) {
 			p += __fsm_sz;
+			parser->state = &&Req_UriMark;
 			__FSM_EXIT(TFW_POSTPONE);
 		}
 		if (__fsm_n < 0)
 			TFW_PARSER_BLOCK(Req_UriMark);
 
-		parser->_i_st = I_0;
+		parser->_i_st = I_ST(I_0);
```

Comment:
We can use the whole body of `__FSM_START_ALT()` in `__parse_http_date()`, but it'd be ugly. `I_ST(I_0)` is also used in `tfw_http_parse_req()`. I thought about the version of the code, but in case of
```C
 if (parser->_i_st == I_ST(I_EoL))
    goto I_EoL;
__FSM_START(parser->_i_st);
```
we won't be able to rping debug message about current state.